### DISCUSSION TITLE
Changing from openddsharp.standard.idlgenerator to openddsharp.idlgenerator

### DISCUSSION THREAD

**Discussion by Oles7** [jmmorato/openddsharp#228] [General]
I have a VS solution that is using openddsharp to build the TypeSupport.cs file so I can use the idl in my projects.

The project was working fine using openddsharp.standard, but since that is deprecated, I am switching over to use the openddsharp nuget package instead.

Now whenever I build my .idl file, the TypeSupport file is not placed in the top directory like it used to be and VS is unable to find the new location of it.

Any ideas why this would be different changing from openddsharp.standard to openddssharp?

**Comment by jmmorato** [comment]
Well... I have made several changes on the project structure and references required, surely you will face many changes if you try to adapt manually.

My advice:
- Backup your IDL file in a different location
- Install the [OpenDDSharp.Templates](https://www.nuget.org/packages/OpenDDSharp.Templates) for your current version of OpenDDSharp
- Replace the IDL file and Rebuild the project again

**Reply by Oles7** [reply]
Should I use the OpenDDSharp - IDL Project Template in VS? Or just manually build and add to VS Solution?

**Reply by jmmorato** [reply]
It should work with VS... but I use to do it with `dotnet new` and add it to the solution after.

You can find an example in the [Getting Started Article](https://www.openddsharp.com/articles/getting_started.html)

**Comment by Oles7** [comment]
Do projects using OpenDDSharp need to target a certain .net framework?

**Reply by jmmorato** [reply]
The library is tested against the following frameworks:

```xml
<TargetFrameworks>net462;net47;net471;net472;net48;net6.0;net7.0;net8.0;</TargetFrameworks>
```
Previous .NET Framework versions should work as well but not tested anymore in the CI pipeline.

I add/remove frameworks based on the [Microsoft Components Lifecycle](https://learn.microsoft.com/en-us/lifecycle/products/microsoft-net-framework)

**Reply by Oles7** [reply]
Thanks. I am currently using it in a project targeting .NET Framework 4.6.1. So would I need to upgrade my target framework to be .NET Framework 4.7.2? The code builds but when I run it, I am getting an error: Could not load file or assembly 'System.Buffers, Version=4.0.2.0'. Would updating the project to .NET 4.7.2 fix this? I am currently using System.Buffers Version 4.5.1.

**Reply by jmmorato** [reply]
Have you tried adding a binding redirect? Or active the autogenerated binding redirect option?
